### PR TITLE
Allow ENV injection into swagger yaml [BW-861]

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala [hunk 1]
```diff
@@ -127,15 +129,30 @@ trait SwaggerResourceHttpService {
...
 
+
+
   /**
    * @return A route that returns the swagger resource.
    */
   final def swaggerResourceRoute: Route = {
     val swaggerDocsDirective = path(separateOnSlashes(swaggerDocsPath))
+
+    def injectBasePath(basePath: Option[String])(response: HttpResponse): HttpResponse = {
```
Comment (cahrens) [inline engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala:140]: Is it possible to add unit test coverage for this?

**Review by salonishah11 — COMMENTED** [review]
The code LGTM. Will give the official 👍  once tests are added.

**Comment by salonishah11** [comment]
@kpierre13 It looks like framework for test for this file already exists. You should be able to add new test here - [SwaggerUiHttpServiceSpec](https://github.com/broadinstitute/cromwell/blob/develop/engine/src/test/scala/cromwell/webservice/SwaggerUiHttpServiceSpec.scala)

File: engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala [hunk 2]
```diff
@@ -127,15 +134,29 @@ trait SwaggerResourceHttpService {
    */
   protected def swaggerDocsPath = s"$swaggerDirectory/$swaggerServiceName.$swaggerResourceType"
 
+
```
Comment (cahrens) [inline engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala:137]: Can you remove these blank lines?

File: engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala [hunk 3]
```diff
@@ -94,6 +96,11 @@ trait SwaggerUiConfigHttpService extends SwaggerUiHttpService {
  * swagger UI, but defaults to "yaml". This is an alternative to spray-swagger's SwaggerHttpService.
  */
 trait SwaggerResourceHttpService {
+
+  def getBasePathOverride(): Option[String] ={
```
Comment (cahrens) [inline engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala:100]: Did you look into if there is a way to mock environment variables in the unittest? In Java I think there is something like "withEnvironmentVariables", but I don't know what is available in the Scala tests.

Comment (cahrens) [reply engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala:100]: I've looked and haven't found anything.

File: engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala [hunk 4]
```diff
@@ -94,6 +96,11 @@ trait SwaggerUiConfigHttpService extends SwaggerUiHttpService {
...
+   Option(System.getenv("SWAGGER_BASE_PATH"))
+  }
```
Comment (cahrens) [inline engine/src/main/scala/cromwell/webservice/SwaggerUiHttpService.scala:102]:
```suggestion
  def getBasePathOverride(): Option[String] = {
    Option(System.getenv("SWAGGER_BASE_PATH"))
  }
```
Cleaned up the spacing/indenting a bit.

**Review by cahrens — APPROVED** [review]
I had a suggestion to clean up the spacing/indenting, but otherwise this LGTM.

**Review by kpierre13 — APPROVED** [review]

**Pull request merged**